docs/how-to/migrate-to-mas.md

```diff
@@ -0,0 +1,145 @@
+
+## Backup the existing synapse charm and restore on the new synapse charm
+### Backup and restore using the s3-integrator charm
+To migrate everything persisted in the Synapse's `/data` directory, we will use the `s3-integrator` charm. For more information please look at the ["Backup and Restore"](https://discourse.charmhub.io/t/synapse-docs-backup-and-restore/13212) section of the charm's documentation. First we deploy the `s3-integrator` charm and configure it with the credentials for the target s3 bucket:
+```
+juju deploy s3-integrator --channel edge
+juju config s3-integrator endpoint=<s3 endpoint> bucket=<bucket name> path=<optional-path> region=<region> s3-uri-style=<path or host>
+juju run s3-integrator/leader sync-s3-credentials access-key=<access-key> secret-key=<secret-key>
+```
+
+Then we integrate with the existing synapse charm and create the backup
+```
+juju relate s3-integrator server
+juju run server/leader create-backup
+```
+
+Take note of the backup ID, we will use it to restore the backup on the new synapse charm.
+```
+juju relate s3-integrator server-mas
+juju run synapse/leader restore-backup backup-id=<backup-id>
+```
+
+## Synapse database migration
+### Stop all services on the new and the existing synapse charm
+```
+juju ssh --container synapse server-mas/0 "pebble stop synapse; pebble stop synapse-mas; pebble stop stats-exporter"
+juju ssh --container synapse server/0 "pebble stop synapse; pebble stop stats-exporter"
+```
+
+### Connect to the database
+```
+juju ssh --container synapse server-mas/0 bash
+```
+
+The following commands assume that you are in the `synapse` container of the `server-mas` application.
+```
+apt update
+apt install wget postgresql-client -y
+wget https://github.com/mikefarah/yq/releases/download/v4.44.3/yq_linux_amd64 -O /usr/bin/yq && chmod +x /usr/bin/yq
+```
+
+Copy the original synapse database
+```
+DB_HOST=$(yq e '.database.args.host' /data/homeserver.yaml)
+DB_PORT=$(yq e '.database.args.port' /data/homeserver.yaml)
+DB_PASSWORD=$(yq e '.database.args.password' /data/homeserver.yaml)
+DB_USER=$(yq e '.database.args.user' /data/homeserver.yaml)
+PGPASSWORD=$DB_PASSWORD psql --host $DB_HOST --username $DB_USER --port $DB_PORT postgres -c 'drop database "server-mas";'
+PGPASSWORD=$DB_PASSWORD psql --host $DB_HOST --username $DB_USER --port $DB_PORT postgres -c "create database \"server-mas\" with template server owner $DB_USER;"
+```
+
+In case where postgres gives you an error saying "", drop all connections from the database
+```
+PGPASSWORD=$DB_PASSWORD psql --host $DB_HOST --username $DB_USER --port $DB_PORT postgres -c SELECT pg_terminate_backend(pg_stat_activity.pid) FROM pg_stat_activity WHERE pg_stat_activity.datname = "server" AND pid <> pg_backend_pid();
+```
+
+## Migrating users to MAS
+### Perform user migration to MAS database
```

Comment:
a note about proxy?

Reply:
I thought about it but I don't think that it should be put here since it's quite specific to the environment, what do you think?

Reply:
Ok, if needed can be added later